### PR TITLE
RavenDB-13760 Sodium & Pal modified to catch all exception on dll load and return detailed message

### PR DESCRIPTION
**Pull request by haludi**
https://issues.hibernatingrhinos.com/issue/RavenDB-13760

### CODE REVIEW

**Review by ayende — APPROVED** [review]

File: src/Sparrow.Server/Platform/Pal.cs [hunk 1]
```diff
@@ -80,17 +80,17 @@ static Pal()
...
-                throw;
-            }
             catch (Exception ex)
             {
-                var ErrString =
-                    "'Microsoft Visual C++ 2015 Redistributable Package' (or newer). It can be downloaded from https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads";
-                throw new IncorrectDllException(
-                    $"{LIBRVNPAL} version might be invalid or not usable on current platform. Initialization error could also be caused by missing {ErrString}",
-                    ex);
+                var platformStr = $"Arch:{RuntimeInformation.OSArchitecture}, OSDesc:{RuntimeInformation.OSDescription}";
```
Comment (ppekrol) [inline src/Sparrow.Server/Platform/Pal.cs]: Missing space after `Arch:` and `OSDesc:`. Do we really need that? It produces longer exception method and the assumption here is what? That the user does not know what OS he is running on? Same info is available in welcome message.

Comment (ppekrol) [reply src/Sparrow.Server/Platform/Pal.cs]: I would suggest to remove it so the message would be more clear.

File: src/Sparrow.Server/Platform/Pal.cs [hunk 2]
```diff
@@ -80,17 +80,17 @@ static Pal()
...
+
+                var errString = $"{LIBRVNPAL} version might be invalid, missing or not usable on current platform '${platformStr}";
```
Comment (ppekrol) [inline src/Sparrow.Server/Platform/Pal.cs]: Should be `current platform (${platformStr}).`

File: src/Sparrow.Server/Sodium.cs [hunk 3]
```diff
@@ -67,32 +68,19 @@ static Sodium()
...
-            catch (DllNotFoundException dllNotFoundEx)
+            catch (Exception ex)
             {
-                // make sure the lib file is not there (this exception might pop when incorrect libsodium lib is does exists)
-                if (File.Exists(LIBSODIUM))
-                {
-                    throw new IncorrectDllException(
-                        $"{LIBSODIUM} probably contains the wrong version or not usable on the current platform. Try installing {LIBSODIUM} and re-run server.",
-                        dllNotFoundEx);
+                var platformStr = $"Arch:{RuntimeInformation.OSArchitecture}, OSDesc:{RuntimeInformation.OSDescription}";
```
Comment (ppekrol) [inline src/Sparrow.Server/Sodium.cs]: Same comments as in Pal.

File: src/Sparrow.Server/Sodium.cs [hunk 4]
```diff
@@ -67,32 +68,19 @@ static Sodium()
...
 
-                }
-            }
+                var errString = $"{LIBSODIUM} version might be invalid, missing or not usable on current platform '${platformStr}";
```
Comment (ppekrol) [inline src/Sparrow.Server/Sodium.cs]: Same comments as in Pal.

File: src/Sparrow.Server/Platform/Pal.cs (hunk 1 above)
Comment (haludi) [reply src/Sparrow.Server/Platform/Pal.cs]: This is for us so if we get this message from a client we don't need to do a ping pong with the client for this information

Comment (ppekrol) [reply src/Sparrow.Server/Platform/Pal.cs]: So it should be at the end of the message. It hides the main part.

Comment (haludi) [reply src/Sparrow.Server/Platform/Pal.cs]: O.k.

**Pull request merged**